src/Services/EnturSales.php

```diff
@@ -54,66 +42,61 @@ public static function dateTimeFormatter($date)
-
-            $matches = null;
-            // match for date on format YYYY-MM-DD, YYYY/MM/DD, YYYY.MM.DD or YYYYMMDD
-            // matches years in range (inclusive) 2000-3999
-            preg_match(
-                '/[^\d](?<date>((2|3)\d{3})([-\/\.])*(0[1-9]|1[0,1,2])([-\/\.])*(0[1-9]|[12][0-9]|3[01]))/',
-                $fileName,
-                $matches
-            );
-
-            if (!isset($matches['date'])) {
-                return null;
-            }
-
-            // Fecth date match from preg_match, and make it same format as $chunkDate* variables
-            $fileDate = preg_replace('/[-\/\.]/', '', $matches['date']);
-
-            // Checking for either chunkId - 1day ($chunkDate) or chunkId (today)
-            // since there seems to be a tiny discrepency in when files are made
-            // available for download.
-            if (strcmp($chunkDate, $fileDate) == 0 || strcmp($chunkDatePrevious, $fileDate) == 0) {
-                $archive = new ChunkArchive(SinkEnturSales::$id, $chunkId);
-                $archive->addFromString($fileName, $response->body());
-                $archive->save();
-
-                $this->debug("Saving file %s, %s", $fileDate, $chunkDate);
-            } else {
-                $response = Http::withHeaders(['authorization' => 'Bearer ' . EnturCleosApi::getApiToken()])
-                    ->get($this->getCleosS1Url($chunkId, $reportID));
-                $status = $response->status();
-            }
+        $client = Http::withHeaders(['authorization' => 'Bearer ' . EnturCleosApi::getApiToken()]);
+        //$response = $client->get(EnturCleosApi::getNextDatasetInProduct(0, $chunkId));
+        //$nextDatasetResult = $client->get(EnturCleosApi::getNextDatasetInProductUrl(0, '2024-12-01'));
+        $nextDatasetResult = $client->get(EnturCleosApi::getNextDatasetInProductUrl(0, $chunkId));
+        $status = $nextDatasetResult->status();
+        if ($status != 200) {
+            $this->debug("STATUS nextDataset: %s", $status);
+            return null;
         }
 
-        if (is_null($archive)) {
-            $this->debug("NO report found for chunkId: %s", $chunkId);
-            return $archive;
+        // next dataset id
+        $datasetId = (int)$nextDatasetResult->body();
+
+        $datasetResult = $client->get(EnturCleosApi::getDatasetUrl($datasetId));
+        $datasetStatus = $datasetResult->status();
+        if ($datasetStatus != 200) {
+            $this->debug("STATUS dataset: %s", $datasetStatus);
+            return null;
         }
 
+        $createReportResult = $client->post(EnturCleosApi::getReportCreateUrl($datasetId));
+        if ($createReportResult->status() != 201) {
+            $this->debug("STATUS createReport: %s", $createReportResult->status());
+            return null;
+        }
+
+        // Poll to check if report is completed
+        $createReportBody = json_decode($createReportResult->body());
```

Comment:
Not a PHP god, but this might work?

```suggestion
        $createReportBody = $createReportResult->json();
```

Reply:
If the $response object has a json function, sure